src/ex3/main-commander.js

```diff
@@ -0,0 +1,86 @@
+import chalk from "chalk";
+import { LogPokemonAscii } from "./pokemon-ascii.js";
+
+export class MainCommander {
+  constructor(itemManagerCommander, pokemonClient) {
+    this.itemManagerCommander = itemManagerCommander;
+    this.pokemonClient = pokemonClient;
+  }
+
+  init() {
+    this.updateTodos(this.itemManagerCommander.init());
+  }
+
+  showTodos() {
+    this.showAmount();
+    this.todos.forEach((todo, index) => {
+      const todoColor = todo.isNew ? 'green' : 'white';
+      if (todo.isNew) this.markTodoAsOld(todo);
+      console.log(chalk.yellow(index), chalk[todoColor](todo.text));
+    });
+  }
+
+  showAmount() {
+    const amount = this.todos.length;
+    if (amount === 1) {
+      console.log(chalk.yellow(`You have only one todo:`));
+    } else {
+      console.log(chalk.yellow(`You have ${amount} todos`));
+    }
+  }
+
+  markTodoAsOld(todo) {
+    this.updateTodos(this.itemManagerCommander.markItemAsOld(todo));
+  }
+
+  addTodo(text) {
+    console.log(chalk.green(`Adding '${text}'`));
+    if (this.pokemonClient.isPokemon(text)) {
+      return this.addPokemon(text.toLowerCase());
+    } else {
+      const isTextNaN = text.split(',').map( el => isNaN(el));
+      if (isTextNaN.includes(true)) {
+        this.updateTodos(this.itemManagerCommander.addItem(text));
+      } else {
+        return this.addPokemon(text);
+      }
+    }
+  }
+
+  updateTodos(updatedArray) {
+    this.todos = updatedArray;
```

Comment:
it's better to use a single source of truth for the data in order to avoid out-of-sync bugs and the need to sync it in the first place. you can use the itemManagerCommander to fetch the todo list instead

Reply:
Do you mean it's not nice to store todos both in a file and in an array?